File: mage/src/main/java/mil/nga/giat/mage/coordinate/CoordinateSystem.java
package mil.nga.giat.mage.coordinate;

public enum CoordinateSystem {
    WGS84(0),
    MGRS(1);

    private int preferenceValue;

    CoordinateSystem(int preferenceValue) {
        this.preferenceValue = preferenceValue;
    }

    public int getPreferenceValue() {
        return preferenceValue;
    }

    public static CoordinateSystem get(int preferenceValue) {
        for(CoordinateSystem coordinateSystem : values()) {
            if(coordinateSystem.preferenceValue == preferenceValue) {
                return coordinateSystem;
            }
        }

        return null;
    }
}

File: mage/src/main/java/mil/nga/giat/mage/coordinate/CoordinateSystem.kt
package mil.nga.giat.mage.coordinate

enum class CoordinateSystem(val preferenceValue: Int) {
   WGS84(0), MGRS(1), DMS(2), GARS(3);

   companion object {
      fun fromPreference(preferenceValue: Int): CoordinateSystem {
         for (coordinateSystem in values()) {
            if (coordinateSystem.preferenceValue == preferenceValue) {
               return coordinateSystem
            }
         }

         return WGS84
      }
   }
}